Kernel/Theory.cpp

```diff
@@ -1249,33 +1197,24 @@ std::string Theory::getInterpretationName(Interpretation interp) {
   }
 }
 
-OperatorType* Theory::getArrayOperatorType(TermList arraySort, Interpretation i) {
-  ASS(arraySort.isArraySort());
-
-  TermList indexSort = SortHelper::getIndexSort(arraySort);
-  TermList innerSort = SortHelper::getInnerSort(arraySort);
-
-  switch (i) {
-    case Interpretation::ARRAY_SELECT:
-      return OperatorType::getFunctionType({ arraySort, indexSort }, innerSort);
-
-    case Interpretation::ARRAY_BOOL_SELECT:
-      return OperatorType::getPredicateType({ arraySort, indexSort });
-
-    case Interpretation::ARRAY_STORE:
-      return OperatorType::getFunctionType({ arraySort, indexSort, innerSort }, arraySort);
-
-    default:
-      ASSERTION_VIOLATION;
-  }
-}
-
 /**
  * Return type of the function representing interpreted function/predicate @c i.
  */
-OperatorType* Theory::getNonpolymorphicOperatorType(Interpretation i)
+OperatorType* Theory::getOperatorType(Interpretation i)
 {
-  ASS(!isPolymorphic(i));
+  if (i == Interpretation::EQUAL) {
+    return OperatorType::getPredicateType(2);
```

Comment:
I would expect either `![A: $tType]: A * A > $o` or a family of `A * A > $o` for each A. Why does this (which I think is `$i * $i > $o`) work?

Reply:
Ahah - so hopefully this `OperatorType` is never used and equality remains a special case. Got it!